pkgs/top-level/haskell-packages.nix

```diff
@@ -72,6 +72,10 @@ in {
         else
           packages.ghc865Binary;
       inherit (buildPackages.python3Packages) sphinx;
+      # Need to use apple's patched xattr until
+      # https://github.com/xattr/xattr/issues/44 and
+      # https://github.com/xattr/xattr/issues/55 are solved.
+      inherit (buildPackages.darwin) xattr;
```

Comment:
Do the Linux builds somehow know not to take things from `buildPackages.darwin`?

Reply:
Due to lazy evaluation, the linux builds won't depend on nor evaluate `xattr`. The `darwin` package set exists on all plaforms, so that's no problem.